module/apmnegroni/middleware.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmnegroni
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/urfave/negroni"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmhttp"
+	"go.elastic.co/apm/stacktrace"
+)
+
+func init() {
+	stacktrace.RegisterLibraryPackage(
+		"github.com/urfave/negroni",
+	)
+}
+
+// Middleware returns a new negroni middleware handler for tracing
+// requests and reporting errors.
+//
+// This middleware will recover and report panics, so it can
+// be used instead of the standard negroni.Recovery middleware.
+//
+// By default, the middleware will use apm.DefaultTracer.
+// Use WithTracer to specify an alternative tracer.
+func Middleware(o ...Option) negroni.Handler {
+	m := &middleware{
```

Comment:
Make this a non-pointer type (`m := middleware{...}`)? The ServeHTTP method is on the non-pointer type middleware